save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/ProjectView.kt

```diff
@@ -396,15 +411,13 @@ class ProjectView : AbstractView<ProjectExecutionRouteProps, ProjectViewState>(f
                     .forEachIndexed { i, projectMenu ->
                         li {
                             className = ClassName("nav-item")
-                            val classVal =
-                                    if ((i == 0 && state.selectedMenu == null) || state.selectedMenu == projectMenu) " active font-weight-bold" else ""
+                            val classVal = if ((i == 0 && state.selectedMenu == null) || state.selectedMenu == projectMenu) " active font-weight-bold" else ""
                             p {
                                 className = ClassName("nav-link $classVal text-gray-800")
                                 onClick = {
                                     if (state.selectedMenu != projectMenu) {
-                                        setState {
-                                            selectedMenu = projectMenu
-                                        }
+                                        changeUrl(projectMenu, ProjectMenuBar)
+                                        setState { selectedMenu = projectMenu }
```

Comment:
I would assume that `changeUrl` should be called whenever `selectedMenu` changes. Can it be moved to something like `componentDidUpdate` so that we won't need to call it manually near every state change?

Reply:
Done